src/ui/ui_bundles/ui_bundles_controller.js

```diff
@@ -110,8 +110,10 @@ export class UiBundlesController {
     return resolve(this._workingDir, ...args);
   }
 
-  getCachePath() {
-    return this.resolvePath('../.cache', this.hashBundleEntries());
+  getCacheDirectory(...subPath) {
+    if (this.isDevMode()) {
```

Comment:
Should we add a comment that explains the reasoning here? Maybe something along the lines of

> Disabling the cache-loader in the optimizer when running in production mode, as it creates cache files in optimize/.cache that are not necessary for distributable versions of Kibana and just make compressing and extracting it more difficult.

That should hopefully make it easier to remember later on why this was works the way it does

Reply:
I appreciate your observation that the relationship between this and the cache loader was not clear, so I think I did a bit better than a comment and put the condition in `maybeAddCacheLoader()` within `BaseOptimizer#getConfig()` instead.